### DISCUSSION TITLE
Best practices with multiple windows?

### DISCUSSION THREAD

**Discussion by tleyden** [tauri-apps/tauri#9423] [Q&A]
I'm creating an OSX Tauri + Vue.js app with multiple windows that will be accessible from the menu bar, like so:

<img width="177" alt="Screenshot 2024-04-09 at 3 10 57 PM" src="https://github.com/tauri-apps/tauri/assets/296876/a21a5985-c429-4241-a208-731c001c87aa">

Here "Show window" will just show the main window that corresponds to `index.html`.  The "Config" window will be a completely separate window.

When adding the conf window for example, I added the following additional files:

* `index_config.html` (copy/paste from `index.html`)
* `main_config.ts` (copy/paste from `main.ts`)
* `AppConfig.vue` (copy/paste from `App.vue`)

Is this what people normally do for each new window being added in their Tauri app?  Are there any examples of multi-window Tauri apps with source code I should look at?

**Comment by raphaelmenges** [comment]
I am facing [a very similar question](https://github.com/tauri-apps/tauri/discussions/11643). In Electron, one can create sub-windows from JavaScript via `window.open()` that have the same context as the parent window. Then you can have a single parent window that owns the Vue context and sub windows that are in the very same Vue context. But this does not seem to be possible in Tauri.

What did you end up doing @tleyden?
